citesphere/src/main/java/edu/asu/diging/citesphere/core/zotero/impl/ZoteroManager.java

```diff
@@ -256,24 +257,62 @@ private CitationResults createCitationResults(ZoteroResponse<Item> response) {
     @Override
     public ICitation updateCitation(IUser user, String groupId, ICitation citation)
             throws ZoteroConnectionException, ZoteroHttpStatusException {
-        Item item = itemFactory.createItem(citation, new ArrayList<>());
+        Item item = itemFactory.createItem(citation, citation.getCollections());
 
         List<String> itemTypeFields = getItemTypeFields(user, citation.getItemType());
         // add fields that need to be submitted
         itemTypeFields.add(ZoteroFields.VERSION);
         itemTypeFields.add(ZoteroFields.ITEM_TYPE);
         itemTypeFields.add(ZoteroFields.CREATOR);
+        
 
         List<String> ignoreFields = createIgnoreFields(itemTypeFields, item, false);
-        ignoreFields.add("collections");
 
-        List<String> validCreatorTypes = getValidCreatorTypes(user, citation.getItemType());
+        if (citation.getCollections() != null) {
+            ignoreFields.remove("collections");
+        }
 
+        List<String> validCreatorTypes = getValidCreatorTypes(user, citation.getItemType());
         Item updatedItem = zoteroConnector.updateItem(user, item, groupId, new ArrayList<>(), ignoreFields,
                 validCreatorTypes);
         return citationFactory.createCitation(updatedItem);
     }
+    
+    /**
+     * We are updating only collections field of a citation. Zotero only requires
```

Comment:
What does that mean "We are updating only collections field of a citation"? This method is used to update multiple items, isn't it? It's updating all the given fields. key, version, and collections are just the only ones that are required?

Reply:
This method can only update the `collections` field of a citation. In order to do that I need to pass the required fields `key and version` in addition to collections. 

Maybe `batchUpdateCollectionOnCitation` might be a good name for this method. Do you want me to change the method name and update the comments to make it more clear?